src/Symfony/Component/Security/Core/Authorization/Voter/Voter.php

```diff
@@ -30,8 +30,16 @@ public function vote(TokenInterface $token, $subject, array $attributes)
         $vote = self::ACCESS_ABSTAIN;
 
         foreach ($attributes as $attribute) {
-            if (!$this->supports($attribute, $subject)) {
-                continue;
+            try {
+                if (!$this->supports($attribute, $subject)) {
+                    continue;
+                }
+            } catch (\TypeError $e) {
+                if (preg_match((sprintf('/Argument 1 passed to ([\w\\\]+)::supports\(\) must be of the type string, (\w+) given/')), $e->getMessage())) {
```

Comment:
```suggestion
                if (0 === strpos($e->getMessage(), 'Argument 1 passed to') && false !== strpos($e->getMessage(), '::supports() must be of the type string')) {
```

Strpos seems to be about 50% quicker: https://3v4l.org/47iXF

This method is potentially called quite a lot (for every voter in access_control and all isGranted calls), so it's performance critical, so I think it's worth considering.

Reply:
Indeed, this is to be considered. Your test is less strict but it is probably sufficient, but above all faster. I make the modification.

Reply:
It is done.

Reply:
@wouterj Please keep in mind that the error message format changes with php 8. The error message does not begin with `Argument` anymore, but with the method name, IIRC.

Reply:
Travis agrees:

```
TypeError: Symfony\Component\Security\Core\Tests\Authorization\Voter\VoterTest_Voter::supports(): Argument #1 ($attribute) must be of type string, stdClass given, called in /home/travis/build/symfony/symfony/src/Symfony/Component/Security/Core/Authorization/Voter/Voter.php on line 34
```

So it probably should do something like:

```php
if (\PHP_VERSION_ID < 80000) {
    if (0 === strpos($e->getMessage(), 'Argument 1 passed to') && false !== strpos($e->getMessage(), '::supports() must be of the type string')) {
        continue;
    }
} else {
    if (false !== strpos($e->getMessage(), 'supports(): Argument #1')) {
        continue;
    }
}

throw $e;
```

Reply:
@wouterj Thank you, I didn't have PHP 8 on hand right now. I take your code.

Reply:
I think almost nobody has. Fortunately, the Symfony test suite runs on PHP 8. You can find its results in the "Travis" link below. The relevant failure is https://travis-ci.org/github/symfony/symfony/jobs/699831814#L4276 (you can ignore the other failures, lots of things are work-in-progress for PHP 8/external dependency support).

Reply:
It is done.